webapp/src/modules/mortgage/sagas.js

```diff
@@ -0,0 +1,143 @@
+  fetchActiveParcelMortgagesFailure
+} from './actions'
+import { getAddress } from 'modules/wallet/selectors'
+import { toInterestRate, getLoanMetadata, daysToSeconds } from './utils'
+import { getKyberOracleAddress } from 'modules/wallet/utils'
+import { locations } from 'locations'
+
+export function* mortgageSaga() {
+  yield takeLatest(CREATE_MORTGAGE_REQUEST, handleCreateMortgageRequest)
+  yield takeLatest(CANCEL_MORTGAGE_REQUEST, handleCancelMortgageRequest)
+  yield takeLatest(FETCH_MORTGAGED_PARCELS_REQUEST, handleFetchMortgageRequest)
+  yield takeLatest(
+    FETCH_ACTIVE_PARCEL_MORTGAGES_REQUEST,
+    handleFetchActiveParcelMortgagesRequest
+  )
+}
+
+function* handleFetchMortgageRequest(action) {
+  try {
+    const { borrower } = action
+    const [parcels, mortgages] = yield all([
+      call(() => api.fetchMortgagedParcels(borrower)),
+      call(() => api.fetchMortgagesByBorrower(borrower))
+    ])
+    yield put(fetchMortgagedParcelsSuccess(parcels, mortgages))
+  } catch (error) {
+    yield put(fetchMortgagedParcelsFailure(error.message))
+  }
+}
+
+function* handleCreateMortgageRequest(action) {
+  try {
+    let {
+      duration,
+      payableAt,
+      expiresAt,
+      interestRate,
+      punitoryRate,
+      parcel,
+      amount
+    } = action
+
+    const rcnEngineContract = eth.getContract('RCNEngine')
+    const mortgageCreatorContract = eth.getContract('MortgageCreator')
+    const landRegistryContract = eth.getContract('LANDRegistry')
+    const kyberOrcaleAddress = getKyberOracleAddress()
+    const manaCurrency = eth.utils.fromAscii('MANA')
+
+    const landId = yield call(() =>
+      landRegistryContract.encodeTokenId(parcel.x, parcel.y)
+    )
+    const borrower = yield select(getAddress)
+    const loanMetadata = getLoanMetadata()
+
+    let loanParams = [
+      eth.utils.toWei(amount), // Amount requested
+      toInterestRate(interestRate), // Anual interest
+      toInterestRate(punitoryRate), // Anual punnitory interest
+      daysToSeconds(duration), // Duration of the loan, in seconds
+      daysToSeconds(payableAt), // Time when the payment of the loan starts
+      expiresAt // Expiration timestamp of the request
+    ]
+
+    // Retrieve the loan signature
+    let loanIdentifier = yield call(() =>
+      rcnEngineContract.buildIdentifier(
+        kyberOrcaleAddress, // Contract of the oracle
+        borrower, // Borrower of the loan (caller of this method)
+        mortgageCreatorContract.address, // Creator of the loan, the mortgage creator
+        manaCurrency, // Currency of the loan, MANA
+        ...loanParams,
+        loanMetadata // Metadata
+      )
+    )
+
+    // Sign the loan
+    let { signature } = yield call(() => eth.sign(loanIdentifier))
+    const approveSignature = signature.slice(2)
+    let r = `0x${approveSignature.slice(0, 64)}`
+    let s = `0x${approveSignature.slice(64, 128)}`
+    let v = eth.utils.toDecimal(`0x${approveSignature.slice(128, 130)}`)
```

Comment:
Houston we (may) have a problem here. Checking the implementation on `decentraland-eth` it looks like `LedgerWallet#sign` has a different result from `NodeWallet#sign`.

Have you tried it with a ledger? If you do have one, remember that it only works in https! so you may want to use https://ngrok.com/ to bridge the connection. @abarmat can help you test too.

I would say that, *if the results are incompatible*, we should change `decentraland-eth`s implmentation. But if we're in a hurry we can typecheck the wallet here and make the change in the meantime.